pkg/k8shandler/common.go

```diff
@@ -0,0 +1,491 @@
+						"/usr/share/elasticsearch/probe/readiness.sh",
+					},
+				},
+			},
+		},
+		VolumeMounts: []v1.VolumeMount{
+			v1.VolumeMount{
+				Name:      "elasticsearch-storage",
+				MountPath: "/elasticsearch/persistent",
+			},
+			v1.VolumeMount{
+				Name:      "elasticsearch-config",
+				MountPath: elasticsearchConfigPath,
+			},
+			v1.VolumeMount{
+				Name:      "certificates",
+				MountPath: elasticsearchCertsPath,
+			},
+		},
+		Resources: resourceRequirements,
+	}
+}
+
+func newProxyContainer(imageName, clusterName string) (v1.Container, error) {
+	proxyCookieSecret, err := utils.RandStringBase64(16)
+	if err != nil {
+		return v1.Container{}, err
+	}
+	container := v1.Container{
+		Name:            "proxy",
+		Image:           imageName,
+		ImagePullPolicy: "IfNotPresent",
+		Ports: []v1.ContainerPort{
+			v1.ContainerPort{
+				Name:          "metrics",
+				ContainerPort: 60000,
+				Protocol:      v1.ProtocolTCP,
+			},
+		},
+		VolumeMounts: []v1.VolumeMount{
+			v1.VolumeMount{
+				Name:      fmt.Sprintf("%s-%s", clusterName, "metrics"),
+				MountPath: "/etc/proxy/secrets",
+			},
+			v1.VolumeMount{
+				Name:      "certificates",
+				MountPath: "/etc/proxy/elasticsearch",
+			},
+		},
+		Args: []string{
+			"--https-address=:60000",
+			"--provider=openshift",
+			"--upstream=https://127.0.0.1:9200",
+			"--tls-cert=/etc/proxy/secrets/tls.crt",
+			"--tls-key=/etc/proxy/secrets/tls.key",
+			"--upstream-ca=/etc/proxy/elasticsearch/admin-ca",
+			"--openshift-service-account=elasticsearch",
+			`-openshift-sar={"resource": "namespaces", "verb": "get"}`,
+			`-openshift-delegate-urls={"/": {"resource": "namespaces", "verb": "get"}}`,
+			"--pass-user-bearer-token",
+			fmt.Sprintf("--cookie-secret=%s", proxyCookieSecret),
+		},
+	}
+	return container, nil
+}
+
+func newEnvVars(nodeName, clusterName, instanceRam string, roleMap map[api.ElasticsearchNodeRole]bool) []v1.EnvVar {
+
+	return []v1.EnvVar{
+		v1.EnvVar{
+			Name:  "DC_NAME",
+			Value: nodeName,
+		},
+		v1.EnvVar{
+			Name: "NAMESPACE",
+			ValueFrom: &v1.EnvVarSource{
+				FieldRef: &v1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		v1.EnvVar{
+			Name:  "KUBERNETES_TRUST_CERT",
+			Value: "true",
+		},
+		v1.EnvVar{
+			Name:  "SERVICE_DNS",
+			Value: fmt.Sprintf("%s-cluster", clusterName),
+		},
+		v1.EnvVar{
+			Name:  "CLUSTER_NAME",
+			Value: clusterName,
+		},
+		v1.EnvVar{
+			Name:  "INSTANCE_RAM",
+			Value: instanceRam,
+		},
+		v1.EnvVar{
+			Name:  "HEAP_DUMP_LOCATION",
+			Value: heapDumpLocation,
+		},
+		v1.EnvVar{
+			Name:  "RECOVER_AFTER_TIME",
+			Value: "5m",
+		},
+		v1.EnvVar{
+			Name:  "READINESS_PROBE_TIMEOUT",
+			Value: "30",
+		},
+		v1.EnvVar{
+			Name:  "POD_LABEL",
+			Value: fmt.Sprintf("cluster=%s", clusterName),
+		},
+		v1.EnvVar{
+			Name:  "IS_MASTER",
+			Value: strconv.FormatBool(roleMap[api.ElasticsearchRoleMaster]),
+		},
+		v1.EnvVar{
+			Name:  "HAS_DATA",
```

Comment:
Needed? Could we not just create the configmap properly now that nodes in different roles will/should have their own config?

Reply:
we're still using a single configmap for all nodes. not sure that we'd get any benefit from having multiples if a single value would change.